### PR TITLE
Adds circuit USB for reflectors

### PR DESCRIPTION
**Pull request by Blevruz**
## About The Pull Request

Reflectors get USB ports. You can put in an angle and, if the reflector is unlocked, it goes to that angle when the component is triggered.

## Why It's Good For The Game

Fun new uses for circuits! Fun new uses for the trig module!! We love lasers!!!

## Changelog

:cl:
add: Reflectors have been slotted with USB ports.
/:cl:

### CODE REVIEW

**Comment by Time-Green** [comment]
This could be so nutty with some smart maths I love it

**Comment by moocowswag** [comment]
Ive already found that emitters and reflectors make useful traps with circuits, making a turret might not be too hard, i dont think the math will be the hard part, itl be establishing a way to consistently get the entity variable/coords of whoever/whatever you want to target to even begin doing the math.


But that will most likely balance this out and make it the same as most other passive circuit systems (You dont die until your entity variable is uncovered from being silly and walking into a scanner gate and then you die instantly)

**Comment by Blevruz** [comment]
yea imo th

> Ive already found that emitters and reflectors make useful traps with circuits, making a turret might not be too hard, i dont think the math will be the hard part, itl be establishing a way to consistently get the entity variable/coords of whoever/whatever you want to target to even begin doing the math.
> 
> But that will most likely balance this out and make it the same as most other passive circuit systems (You dont die until your entity variable is caught from being silly and walking into a scanner gate and then you die instantly)

yea i think this and the need for line of sight or the fairly-annoying-to-get x ray guns should be interesting obstacles in playing with this.

File: code/game/objects/structures/reflector.dm [hunk 1]
```diff
@@ -256,3 +260,30 @@
 		return
 	else
 		return ..()
+
+//	USB
+
+/obj/item/circuit_component/reflector
+	display_name = "Reflector"
+	desc = "Allows to adjust the angle of a reflector."
```
Comment (LemonInTheDark) [inline code/game/objects/structures/reflector.dm:268]: Allows to -> Allows you to?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/reflector.dm (hunk 1 above)
Comment (Blevruz) [reply code/game/objects/structures/reflector.dm]: yeah that makes sense

**Comment by Lauwma** [comment]
Except we don't have a square root module to convert cartesian into angular coordinates

**Comment by Lauwma** [comment]
> > Except we don't have a square root module to convert cartesian into angular coordinates
> 
> 
> 
> you can just use arctan. tl;dr: tan(_a_) = (length of opposite side) / (length of adjacent side); if you know the position (_rx_, _ry_) of your reflector and the position (_tx_, _ty_) of your target you can get _a_ with arctan( (_rx_ - _tx_) / (_ry_ - _ty_) ) over a 180° arc (can trivially be patched for 360°). give it a shot! its quite fun.
> 
> 

You need R for this which is the distance between your two points and to calculate it you need Pythagorean equation o^2 + a^2 = r^2. To have r from r^2 you need to use sqrt

**Comment by Blevruz** [comment]
> > > Except we don't have a square root module to convert cartesian into angular coordinates
> > 
> > 
> > you can just use arctan. tl;dr: tan(_a_) = (length of opposite side) / (length of adjacent side); if you know the position (_rx_, _ry_) of your reflector and the position (_tx_, _ty_) of your target you can get _a_ with arctan( (_rx_ - _tx_) / (_ry_ - _ty_) ) over a 180° arc (can trivially be patched for 360°). give it a shot! its quite fun.
> 
> You need R for this which is the distance between your two points and to calculate it you need Pythagorean equation o^2 + a^2 = r^2. To have r from r^2 you need to use sqrt

why do you need r?

**Pull request merged**